services/API-service/src/api/event/event-pcode.entity.ts

```diff
@@ -0,0 +1,26 @@
+import { Entity, Column, Check, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity('event_pcode', { schema: 'IBF-pipeline-output' })
+export class EventPcodeEntity {
+  @PrimaryGeneratedColumn()
+  public id: number;
```

Comment:
use uuid for primary column like [here](https://github.com/rodekruis/IBF-system/blob/master/services/API-service/src/api/country/country.entity.ts#L16-L17)

Reply:
I tried that before but I ran into an issue when using a join in one of the raw queries. That a number could not be joined with a string. Do you think it's worth the effort to try and change this?